WordPress/Classes/ViewRelated/Media/Tenor/TenorAPI/TenorClient.swift

```diff
@@ -0,0 +1,80 @@
+// Encapsulates Tenor API Client to interact with Tenor
+
+import Foundation
+import Alamofire
+
+// MARK: Search endpoint & params
+
+private enum EndPoints: String {
+    case search = "https://api.tenor.com/v1/search"
+}
+
+private struct SearchParams {
+    static let key = "key"
+    static let searchString = "q"
+    static let limit = "limit"
+    static let position = "pos"
+}
+
+// MARK: - TenorClient
+
+private struct ClientConfig {
+    var apiKey: String?
+}
+
+struct TenorClient {
+    typealias TenorSearchResult = ((_ data: [TenorGIF]?, _ position: String?, _ error: Error?) -> Void)
+
+    private var config: ClientConfig
+    static var shared = TenorClient()
+
+    private init() {
+        config = ClientConfig()
+    }
+
+    static func configure(apiKey: String) {
+        shared.config.apiKey = apiKey
+    }
+
+    // MARK: - Public Methods
+
+    func search(for query: String, limit: Int = 20, from position: String?, completion: @escaping TenorSearchResult) {
```

Comment:
Since this is the public search method, can we add a comment block describing what it does? There are many examples in the codebase you can reference for how we typically format the comments.

Reply:
Thanks @ScoutHarris. I have added comment for this method (Used `Option + Cmd + /` to ask Xcode to generate the comment template ;-) ).